client/src/main/java/org/eclipse/hono/client/impl/TelemetrySenderImpl.java

```diff
@@ -166,37 +168,68 @@ public boolean send(final Message rawMessage) {
-        addPropterties(msg, deviceId, contentType);
-        send(msg, capacityAvailableHandler);
+        send(deviceId, null, payload, contentType, capacityAvailableHandler);
     }
 
     @Override
     public boolean send(final String deviceId, final String payload, final String contentType) {
+        return send(deviceId, null, payload, contentType);
+    }
+
+    @Override
+    public void send(final String deviceId, final String payload, final String contentType, final Handler<Void> capacityAvailableHandler) {
+        send(deviceId, null, payload, contentType, capacityAvailableHandler);
+    }
+
+    @Override
+    public boolean send(final String deviceId, final Map<String, ?> properties, final String payload, final String contentType) {
         final Message msg = ProtonHelper.message(payload);
-        addPropterties(msg, deviceId, contentType);
+        setApplicationProperties(msg, properties);
+        addProperties(msg, deviceId, contentType);
         return send(msg);
     }
 
     @Override
-    public void send(final String deviceId, final String payload, final String contentType, final Handler<Void> capacityAvailableHandler) {
+    public boolean send(final String deviceId, final Map<String, ?> properties, final byte[] payload, final String contentType) {
+        final Message msg = ProtonHelper.message();
+        msg.setBody(new Data(new Binary(payload)));
+        setApplicationProperties(msg, properties);
+        addProperties(msg, deviceId, contentType);
+        return send(msg);
+    }
+
+    @Override
+    public void send(final String deviceId, final Map<String, ?> properties, final String payload, final String contentType, final Handler<Void> capacityAvailableHandler) {
         final Message msg = ProtonHelper.message(payload);
-        addPropterties(msg, deviceId, contentType);
+        setApplicationProperties(msg, properties);
+        addProperties(msg, deviceId, contentType);
+        send(msg, capacityAvailableHandler);
+    }
+
+    @Override
+    public void send(final String deviceId, final Map<String, ?> properties, final byte[] payload, final String contentType, final Handler<Void> capacityAvailableHandler) {
+        final Message msg = ProtonHelper.message();
+        msg.setBody(new Data(new Binary(payload)));
+        setApplicationProperties(msg, properties);
+        addProperties(msg, deviceId, contentType);
         send(msg, capacityAvailableHandler);
     }
 
-    private void addPropterties(final Message msg, final String deviceId, final String contentType) {
+    private void addProperties(final Message msg, final String deviceId, final String contentType) {
         msg.setMessageId(String.format("TelemetryClientImpl-%d", messageCounter.getAndIncrement()));
         msg.setContentType(contentType);
         addDeviceId(msg, deviceId);
     }
+
+    private void setApplicationProperties(final Message msg, final Map<String, ?> properties) {
```

Comment:
Agree.

As defined in the AMQP 1.0 spec, only simple types are allowed and not lists, maps and arrays. I'll check for them as illegal type (even if it seems that proton allows maps and lists, not arrays).

I'm going to update it ;)